src/python/pants/source/wrapped_globs.py

```diff
@@ -120,6 +125,13 @@ def files_hash(self):
         h.update(f.read())
     return h.digest()
 
+  def iter_relative_paths(self):
+    """An alternative `__iter__` that joins files with the relative root."""
+    for f in self:
+      path_from_buildroot = os.path.join(self.rel_root, f)
+      if matches_filespec(path_from_buildroot, self.filespec):
```

Comment:
Corrected. Pushed `matches` to `FilesetWithSpec`, and removed all the regex logic.